lib/services/presets.go

```diff
@@ -540,6 +540,7 @@ func defaultAllowAccessRequestConditions(enterprise bool) map[string]*types.Acce
 					teleport.PresetAccessRoleName,
 					teleport.PresetGroupAccessRoleName,
 				},
+				MaxDuration: types.NewDuration(maxAccessDuration),
```

Comment:
@mdwn 
Are you certain that this approach will function correctly?

According to the flow, the `defaultAllowAccessRequestConditions` function is invoked from within `createPresetRoles`, and the calculation of `MaxDuration` is based on the requested role, not the role requesting access. You can find the relevant code snippet at: [link](https://github.com/gravitational/teleport/blob/ef6e38a0f011bf432ce62aad685aaf1f326a7093/lib/services/access_request.go#L1254)

As a result, for built-in or preset roles like editor/access, this flow should function correctly (assuming those roles will be regenerated or updated). However, for newly created roles, such as those generated by the OKTA/access list sync flow, `MaxDuration` will remain unset.

Reply:
You're right. I think when I tested this I explicitly set a role with the max duration and got a bit confused here. Update to this PR incoming as soon as I figure out a few details w/r/t updating the existing presets.

Reply:
Actually, looks like I'm wrong here, or rather I was originally right -- this *is* how things work here. The max durations from the user's static roles match here against the roles represented in search_as_roles.